File: games/tictactoe/TicTacToe.cs
using System;
using System.Drawing;
using System.Windows.Forms;

namespace TicTacToe
{
    public partial class TicTacToeForm : Form
    {
        // Constants
        private const int SCREEN_SIZE = 600;
        private const int CELL_SIZE = SCREEN_SIZE / 3;
        private readonly Color BACKGROUND_COLOR = Color.White;
        private readonly Color LINE_COLOR = Color.Black;
        private readonly Color X_COLOR = Color.Blue;
        private readonly Color O_COLOR = Color.Red;
        private const int LINE_WIDTH = 2;
        private const int SYMBOL_WIDTH = 3;
        private const int SYMBOL_MARGIN = 50;
        private const int FONT_SIZE = 20;

        private Board gameBoard;
        private Label statusLabel;

        public TicTacToeForm()
        {
            InitializeComponents();
            gameBoard = new Board();
        }

        private void InitializeComponents()
        {
            this.Text = "Tic Tac Toe";
            this.Size = new Size(SCREEN_SIZE, SCREEN_SIZE + 60); // Extra height for status label
            this.BackColor = BACKGROUND_COLOR;
            this.DoubleBuffered = true;
            this.FormBorderStyle = FormBorderStyle.FixedSingle;
            this.MaximizeBox = false;
            this.StartPosition = FormStartPosition.CenterScreen;

            // Add status label at the bottom
            statusLabel = new Label
            {
                TextAlign = ContentAlignment.MiddleCenter,
                Width = SCREEN_SIZE,
                Height = 30,
                Location = new Point(0, SCREEN_SIZE),
                Font = new Font("Arial", FONT_SIZE)
            };
            this.Controls.Add(statusLabel);

            // Add event handlers
            this.Paint += TicTacToeForm_Paint;
            this.MouseClick += TicTacToeForm_MouseClick;
            this.KeyDown += TicTacToeForm_KeyDown;

            // Set the initial status text
            UpdateStatus();
        }

        private void TicTacToeForm_Paint(object sender, PaintEventArgs e)
        {
            Graphics g = e.Graphics;
            DrawGrid(g);
            DrawSymbols(g);
        }

        private void TicTacToeForm_MouseClick(object sender, MouseEventArgs e)
        {
            if (gameBoard.GameOver)
                return;

            Tuple<int, int> cell = GetCellFromPosition(e.X, e.Y);
            if (cell != null)
            {
                int row = cell.Item1;
                int col = cell.Item2;
                if (gameBoard.MakeMove(row, col))
                {
                    UpdateStatus();
                    this.Invalidate(); // Force redraw
                }
            }
        }

        private void TicTacToeForm_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.R)
            {
                gameBoard.ResetGame();
                UpdateStatus();
                this.Invalidate();
            }
        }

        private Tuple<int, int> GetCellFromPosition(int x, int y)
        {
            if (x < 0 || x >= SCREEN_SIZE || y < 0 || y >= SCREEN_SIZE)
                return null;

            int row = y / CELL_SIZE;
            int col = x / CELL_SIZE;
            if (row >= 0 && row < 3 && col >= 0 && col < 3)
                return new Tuple<int, int>(row, col);

            return null;
        }

        private void DrawGrid(Graphics g)
        {
            using (Pen pen = new Pen(LINE_COLOR, LINE_WIDTH))
            {
                // Draw vertical lines
                for (int x = CELL_SIZE; x < SCREEN_SIZE; x += CELL_SIZE)
                {
                    g.DrawLine(pen, x, 0, x, SCREEN_SIZE);
                }

                // Draw horizontal lines
                for (int y = CELL_SIZE; y < SCREEN_SIZE; y += CELL_SIZE)
                {
                    g.DrawLine(pen, 0, y, SCREEN_SIZE, y);
                }
            }
        }

        private void DrawSymbols(Graphics g)
        {
            for (int row = 0; row < 3; row++)
            {
                for (int col = 0; col < 3; col++)
                {
                    int centerX = col * CELL_SIZE + CELL_SIZE / 2;
                    int centerY = row * CELL_SIZE + CELL_SIZE / 2;

                    if (gameBoard.Grid[row, col] == 'X')
                    {
                        // Draw X
                        using (Pen pen = new Pen(X_COLOR, SYMBOL_WIDTH))
                        {
                            int startX = centerX - CELL_SIZE / 2 + SYMBOL_MARGIN;
                            int endX = centerX + CELL_SIZE / 2 - SYMBOL_MARGIN;
                            int startY = centerY - CELL_SIZE / 2 + SYMBOL_MARGIN;
                            int endY = centerY + CELL_SIZE / 2 - SYMBOL_MARGIN;
                            g.DrawLine(pen, startX, startY, endX, endY);
                            g.DrawLine(pen, startX, endY, endX, startY);
                        }
                    }
                    else if (gameBoard.Grid[row, col] == 'O')
                    {
                        // Draw O
                        using (Pen pen = new Pen(O_COLOR, SYMBOL_WIDTH))
                        {
                            int radius = CELL_SIZE / 2 - SYMBOL_MARGIN;
                            g.DrawEllipse(pen, centerX - radius, centerY - radius, radius * 2, radius * 2);
                        }
                    }
                }
            }
        }

        private void UpdateStatus()
        {
            if (gameBoard.GameOver)
            {
                if (gameBoard.Winner != '\0')
                {
                    statusLabel.Text = $"Player {gameBoard.Winner} wins! Press R to restart";
                }
                else
                {
                    statusLabel.Text = "It's a draw! Press R to restart";
                }
            }
            else
            {
                statusLabel.Text = $"Player {gameBoard.CurrentPlayer}'s turn";
            }
        }
    }

    public class Board
    {
        public char[,] Grid { get; private set; }
        public char CurrentPlayer { get; private set; }
        public char Winner { get; private set; }
        public bool GameOver { get; private set; }

        public Board()
        {
            ResetGame();
        }

        public void ResetGame()
        {
            // Initialize empty 3x3 board
            Grid = new char[3, 3];
            for (int row = 0; row < 3; row++)
            {
                for (int col = 0; col < 3; col++)
                {
                    Grid[row, col] = '\0';
                }
            }
            CurrentPlayer = 'X';
            Winner = '\0';
            GameOver = false;
        }

        public bool IsValidMove(int row, int col)
        {
            return Grid[row, col] == '\0' && !GameOver;
        }

        public bool MakeMove(int row, int col)
        {
            if (IsValidMove(row, col))
            {
                Grid[row, col] = CurrentPlayer;
                if (CheckWinner(row, col))
                {
                    Winner = CurrentPlayer;
                    GameOver = true;
                }
                else if (IsBoardFull())
                {
                    GameOver = true;
                }
                else
                {
                    CurrentPlayer = CurrentPlayer == 'X' ? 'O' : 'X';
                }
                return true;
            }
            return false;
        }

        private bool CheckWinner(int row, int col)
        {
            // Check row
            bool rowWin = true;
            for (int c = 0; c < 3; c++)
            {
                if (Grid[row, c] != CurrentPlayer)
                {
                    rowWin = false;
                    break;
                }
            }
            if (rowWin) return true;

            // Check column
            bool colWin = true;
            for (int r = 0; r < 3; r++)
            {
                if (Grid[r, col] != CurrentPlayer)
                {
                    colWin = false;
                    break;
                }
            }
            if (colWin) return true;

            // Check diagonals
            if (row == col)
            {
                bool diagWin = true;
                for (int i = 0; i < 3; i++)
                {
                    if (Grid[i, i] != CurrentPlayer)
                    {
                        diagWin = false;
                        break;
                    }
                }
                if (diagWin) return true;
            }

            if (row + col == 2)
            {
                bool antiDiagWin = true;
                for (int i = 0; i < 3; i++)
                {
                    if (Grid[i, 2 - i] != CurrentPlayer)
                    {
                        antiDiagWin = false;
                        break;
                    }
                }
                if (antiDiagWin) return true;
            }

            return false;
        }

        private bool IsBoardFull()
        {
            for (int row = 0; row < 3; row++)
            {
                for (int col = 0; col < 3; col++)
                {
                    if (Grid[row, col] == '\0')
                    {
                        return false;
                    }
                }
            }
            return true;
        }
    }

    static class Program
    {
        [STAThread]
        static void Main()
        {
            Application.EnableVisualStyles();
            Application.SetCompatibleTextRenderingDefault(false);
            Application.Run(new TicTacToeForm());
        }
    }
}

File: games/tictactoe/tictactoe.py
import pygame
import sys

# Initialize Pygame
pygame.init()

# Constants
SCREEN_SIZE = 600
CELL_SIZE = SCREEN_SIZE // 3
BACKGROUND_COLOR = (255, 255, 255)  # White
LINE_COLOR = (0, 0, 0)             # Black
X_COLOR = (0, 0, 255)              # Blue
O_COLOR = (255, 0, 0)              # Red
LINE_WIDTH = 2
SYMBOL_WIDTH = 3
SYMBOL_MARGIN = 50
FONT_SIZE = 40

# Create game window
screen = pygame.display.set_mode((SCREEN_SIZE, SCREEN_SIZE))
pygame.display.set_caption("Tic Tac Toe")

class Board:
    def __init__(self):
        self.reset_game()
    
    def reset_game(self):
        # Initialize empty 3x3 board
        self.grid = [['' for _ in range(3)] for _ in range(3)]
        self.current_player = 'X'
        self.winner = None
        self.game_over = False
    
    def get_cell_from_pos(self, pos):
        x, y = pos
        row = y // CELL_SIZE
        col = x // CELL_SIZE
        if 0 <= row < 3 and 0 <= col < 3:
            return row, col
        return None
    
    def is_valid_move(self, row, col):
        return self.grid[row][col] == '' and not self.game_over
    
    def make_move(self, row, col):
        if self.is_valid_move(row, col):
            self.grid[row][col] = self.current_player
            if self.check_winner(row, col):
                self.winner = self.current_player
                self.game_over = True
            elif self.is_board_full():
                self.game_over = True
            else:
                self.current_player = 'O' if self.current_player == 'X' else 'X'
            return True
        return False
    
    def check_winner(self, row, col):
        # Check row
        if all(self.grid[row][c] == self.current_player for c in range(3)):
            return True
        # Check column
        if all(self.grid[r][col] == self.current_player for r in range(3)):
            return True
        # Check diagonals
        if row == col and all(self.grid[i][i] == self.current_player for i in range(3)):
            return True
        if row + col == 2 and all(self.grid[i][2-i] == self.current_player for i in range(3)):
            return True
        return False
    
    def is_board_full(self):
        return all(self.grid[row][col] != '' for row in range(3) for col in range(3))

def draw_grid():
    # Draw vertical lines
    for x in range(CELL_SIZE, SCREEN_SIZE, CELL_SIZE):
        pygame.draw.line(screen, LINE_COLOR, (x, 0), (x, SCREEN_SIZE), LINE_WIDTH)
    
    # Draw horizontal lines
    for y in range(CELL_SIZE, SCREEN_SIZE, CELL_SIZE):
        pygame.draw.line(screen, LINE_COLOR, (0, y), (SCREEN_SIZE, y), LINE_WIDTH)

def draw_symbols(board):
    for row in range(3):
        for col in range(3):
            center_x = col * CELL_SIZE + CELL_SIZE // 2
            center_y = row * CELL_SIZE + CELL_SIZE // 2
            if board.grid[row][col] == 'X':
                # Draw X
                start_x = center_x - CELL_SIZE // 2 + SYMBOL_MARGIN
                end_x = center_x + CELL_SIZE // 2 - SYMBOL_MARGIN
                start_y = center_y - CELL_SIZE // 2 + SYMBOL_MARGIN
                end_y = center_y + CELL_SIZE // 2 - SYMBOL_MARGIN
                pygame.draw.line(screen, X_COLOR, (start_x, start_y), (end_x, end_y), SYMBOL_WIDTH)
                pygame.draw.line(screen, X_COLOR, (start_x, end_y), (end_x, start_y), SYMBOL_WIDTH)
            elif board.grid[row][col] == 'O':
                # Draw O
                radius = CELL_SIZE // 2 - SYMBOL_MARGIN
                pygame.draw.circle(screen, O_COLOR, (center_x, center_y), radius, SYMBOL_WIDTH)

def draw_status(board):
    # Initialize font
    font = pygame.font.Font(None, FONT_SIZE)
    
    if board.game_over:
        if board.winner:
            text = f"Player {board.winner} wins! Press R to restart"
        else:
            text = "It's a draw! Press R to restart"
    else:
        text = f"Player {board.current_player}'s turn"
    
    # Create text surface
    text_surface = font.render(text, True, LINE_COLOR)
    text_rect = text_surface.get_rect(center=(SCREEN_SIZE // 2, SCREEN_SIZE - 30))
    screen.blit(text_surface, text_rect)

def main():
    running = True
    board = Board()
    
    while running:
        # Event handling
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                running = False
            elif event.type == pygame.MOUSEBUTTONDOWN and not board.game_over:
                # Handle mouse clicks
                mouse_pos = pygame.mouse.get_pos()
                cell = board.get_cell_from_pos(mouse_pos)
                if cell:
                    row, col = cell
                    board.make_move(row, col)
            elif event.type == pygame.KEYDOWN:
                if event.key == pygame.K_r:  # Reset game when 'R' is pressed
                    board.reset_game()
            
        # Fill the screen with background color
        screen.fill(BACKGROUND_COLOR)
        
        # Draw the grid
        draw_grid()
        
        # Draw X's and O's
        draw_symbols(board)
        
        # Draw game status
        draw_status(board)
        
        # Update the display
        pygame.display.flip()

    pygame.quit()
    sys.exit()

if __name__ == "__main__":
    main()
